src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeConsecutiveIfStatementsTests.cs

```diff
@@ -0,0 +1,1685 @@
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+            System.Console.WriteLine(a || b);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedIntoParentWithStatementWithoutBlock()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+            System.Console.WriteLine(a || b);
+        else [||]if (b)
+            System.Console.WriteLine(a || b);
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+            System.Console.WriteLine(a || b);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedIntoParentWithDifferenceInBlocks1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+            System.Console.WriteLine(a || b);
+        else [||]if (b)
+        {
+            System.Console.WriteLine(a || b);
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+            System.Console.WriteLine(a || b);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedIntoParentWithDifferenceInBlocks2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            System.Console.WriteLine(a || b);
+        }
+        else [||]if (b)
+            System.Console.WriteLine(a || b);
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+            System.Console.WriteLine(a || b);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedIntoParentWithDifferenceInBlocks3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            System.Console.WriteLine(a || b);
+        }
+        else [||]if (b)
+        {
+            {
+                System.Console.WriteLine(a || b);
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+            System.Console.WriteLine(a || b);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoParentWithUnmatchingStatements1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            System.Console.WriteLine(a || b);
+        }
+        else [||]if (b)
+        {
+            System.Console.WriteLine(a || a);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoParentWithUnmatchingStatements2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+            System.Console.WriteLine(a || b);
```

Comment:
do we have tests with single line ifs and single line elseifs?

Reply:
Yes.

Reply:
Actually no, not for the merging refactorings.

Reply:
➡️ Fixed in 8445048ee66a7cd6425ee86485ea41d895675f88